### ISSUE TITLE
Update `baseUrl` to use environment variables

### ISSUE THREAD

**Issue by francis-robert** [therungg/therun-frontend#37]
Yeah, I saw that baseUrl global and wondered if it shouldn't be switched to using environment variables, but I've yet to test it. Something like this
```
export function getBaseUrl() {
  if (process.env.NEXT_PUBLIC_VERCEL_ENV === "production")
    return "https://therun.gg";
  if (process.env.NEXT_PUBLIC_VERCEL_ENV === "preview")
    return `https://${process.env.NEXT_PUBLIC_VERCEL_URL}`; //Get the generated slug
  return "http://localhost:3000";
}
```

_Originally posted by @dotneB in https://github.com/therungg/therun-frontend/issues/20#issuecomment-1549962895_

**Comment by therungg** [comment]
I would prefer to make this the logic:

- Create an ENV var to store the baseURL: makes it flexible for local url

1. Check if `env.process.BASE_URL` is set. If yes, return that.
2. Check if `process.env.NEXT_PUBLIC_VERCEL_URL` is set. If yes, return that.
3. Some fallback (not sure which one exactly atm)

**Comment by therungg** [comment]
Fixed with #49

**Comment by dotneB** [comment]
Is there a use-case that makes it required to have a fallback that checks `req.headers.host`?
If we removed that fallback, we could move that to a lib function that avoids having to create a context and passing it as a prop
```
const getBaseUrl = () => {
    if (process.env.NEXT_PUBLIC_BASE_URL)
        return process.env.NEXT_PUBLIC_BASE_URL;
    if (process.env.NEXT_PUBLIC_VERCEL_URL)
        return process.env.NEXT_PUBLIC_VERCEL_URL;

    return "http://localhost:3000";
	// or return https:/therun.gg;
	// or throw new Error("You need to define the base url env variable;
};
```
